lib/src/asset/id.dart

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Like the AssetId class from barback but without the barback dependency.
+class AssetId {
```

Comment:
Is it possible to get rid of this entirely? Looking through the remaining uses, it seems like almost all of them exist in contexts where the package is determined by context, so you could just use the executable name.

Reply:
I tried that at first and hit at least one case where we were storing a collection of AssetIds and lost context of their associated packages.

I can take another look, but I'd prefer to do that in a new PR.

Reply:
I overestimated the scope. I replaced `id` with a `packageName` and `pathInPackage` argument in the `snapshot` method, but everything else was, as you pointed out, already in a context where we had the package name.